src/main/java/org/opentripplanner/routing/edgetype/StreetCostCalculator.java

```diff
@@ -0,0 +1,57 @@
+package org.opentripplanner.routing.edgetype;
+
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.api.request.WheelchairAccessibilityRequest;
+import org.opentripplanner.routing.core.TraverseMode;
+
+public interface StreetCostCalculator {
+  boolean isStairs();
+
+  double getMaxSlope();
+
+  boolean isWheelchairAccessible();
+
+  default double addWheelchairCost(double cost, WheelchairAccessibilityRequest wheelchair) {
+    var slopeExceededBy = Math.abs(getMaxSlope()) - wheelchair.maxSlope();
+
+    if (slopeExceededBy > 0.00001) {
+      double reluctance = wheelchair.slopeExceededReluctance();
+      if (reluctance > 0) {
+        // if we exceed the max slope the cost increases multiplied by how much you go over the maxSlope
+        var excessMultiplier = (slopeExceededBy * 1000) * reluctance;
+        cost *= excessMultiplier;
+      }
```

Comment:
What is the unit for the `slope`? I tried to find it in the code, but it i not mentioned in JavaDoc. The GTFS speck say: "Maximum slope ratio of the pathway.  Example: In the US, 0.083 (also written 8.3%) is the maximum slope ratio for hand-propelled wheelchair, which mean an increase of 0.083m (so 8.3cm) for each 1m."

Reply:
`var excessMultiplier = (slopeExceededBy * 1000) * reluctance;` 

I think we should remove the 1000 constant here, and instead increase the default value.

Reply:
> `var excessMultiplier = (slopeExceededBy * 1000) * reluctance;`
> 
> I think we should remove the 1000 constant here, and instead increase the default value.

We could but then you would have to configure a very large reluctance because `slopeExceededBy` is a very small number. For example, going one percent over the limit would be `0.01`. You have to have a reluctance of 100 to not have a penalty at all, otherwise you would actually prefer edges that go over the max slope.

This is very different from all the other reluctances where usually 1 means that there is no penalty.

Are you sure that you want a reluctance of 100 to mean "no extra cost"?

Reply:
A factor should never be < 1, I see that you avoid the problem by not using 0 here: `if (slopeExceededBy > 0.00001)`, but this shifts the limit.

Reply:
I've changed the implemenetation here: https://github.com/ibi-group/OpenTripPlanner/commit/7411c60c2b10c86f93ce76e9cfa0fd4ada2d2ac7

Reply:
We decided to go for:
```
var excessMultiplier = 1 + 100 * slopeExceededBy * reluctance;
```

The if you exceed the limit by one percent the factor doubles if the reluctance is 1.